Tests/SwiftParserTest/AttributeTests.swift

```diff
@@ -63,7 +63,11 @@ final class AttributeTests: XCTestCase {
         DiagnosticSpec(message: "expected ':' in attribute argument", fixIts: ["insert ':'"]),
         DiagnosticSpec(message: "expected ')' to end attribute", fixIts: ["insert ')'"]),
         DiagnosticSpec(message: "expected declaration after attribute", fixIts: ["insert declaration"]),
-      ]
+      ],
+      fixedSource: """
+        @_specialize(e
+        <#declaration#>
+        """
```

Comment:
Ah, you’re already running into the limitations of `FixItApplier`. I think because of the way it rewrites the syntax tree, it’s dropping the Fix-Its for the first two diagnostics. I think we would need to change `FixItApplier` to be completely textual instead of syntax-tree-based for this. I.e. there should be a first pass that just gathers edits to the syntax tree, as specified by the Fix-Its and a second pass will apply those edits to the source code.

Would you be interested in looking into that?